### PR TITLE
Fix crash due to un-checked error in LVReaderHandler::handleArchive method

### PR DESCRIPTION
**Pull request by aurelien35**
Fix crash due to un-checked error in LVReaderHandler::handleArchive method.

Correction thanks to @CarlosAlbertoEnciso suggestion

### CODE REVIEW

**Comment by aurelien35** [comment]
Suggested fix for this issue: https://github.com/llvm/llvm-project/issues/118753

**Comment by CarlosAlbertoEnciso** [comment]
@aurelien35 Can you add a test case that loads a library? Thanks

**Comment by CarlosAlbertoEnciso** [comment]
This is the directory for the `llvm-debuginfo-analyzer` [tests](https://github.com/llvm/llvm-project/tree/main/llvm/test/tools/llvm-debuginfo-analyzer). They are divided into `DWARF`, `COFF` and `WebAssembly`.

You can check the tests for `llvm-dwarfdump` [archive](https://github.com/llvm/llvm-project/blob/main/llvm/test/tools/llvm-dwarfdump/X86/archive.test). There is already an input file.

The test file can be in the `DWARF` directory.
The command line for the test file would include some lines like:

```
; RUN: llvm-debuginfo-analyzer --attribute=format \
; RUN:                         --print=scopes \
; RUN:                         %S/../../dsymutil/Inputs/libfat-test.a 2>&1 | \
; RUN: FileCheck --strict-whitespace %s

; CHECK:      Logical View:
; CHECK-NEXT: [000]           {File} 'libfat-test.a' -> Mach-O 64-bit x86-64
; CHECK-EMPTY:
; CHECK-NEXT: [001]             {CompileUnit} 'fat-test.c'
more lines ...
```

Additional information: [filecheck](https://llvm.org/docs/CommandGuide/FileCheck.html), [llvm testing](https://llvm.org/docs/TestingGuide.html).

Any question, please do let me know. Happy to help.

**Comment by aurelien35** [comment]
I've extended the llvm/unittests/DebugInfo/LogicalView/CodeViewReaderTest.cpp unit test with a unit test to load a library file.

Without the merge request, it fails.
With it, it succeed.

I've made the .lib file using the "lib.exe" tool from msvc on the "llvm/unittests/DebugInfo/LogicalView/Inputs/test-codeview-msvc.o" file.

I'm investigating further for other bugs concerning MSVC static libraries debug info loading.

Do you know who can I contact about the MSVC CodeView information parsing specific topic?

**Comment by CarlosAlbertoEnciso** [comment]
This was the starting documentation I used for the `llvm-debuginfo-analyzer` `CodeView` support.

A presentation done by `Reid Kleckner`
https://llvm.org/devmtg/2016-11/Slides/Kleckner-CodeViewInLLVM.pdf
@rnk

**Comment by rnk** [comment]
Thanks for the fix!

@tstellar @tru : note the use of the binary file for unittest purposes here. I don't see a reasonable way to avoid this file and retain this test with today's test infra system.

**Comment by CarlosAlbertoEnciso** [comment]
Quite interesting your approach to use a unit test to load a MSVC library.

**Comment by tstellar** [comment]
Would it be possible to add a README or something that explains how the file was created?

**Comment by aurelien35** [comment]
@tstellar : I've added  a README.md file which explains how I generated the .lib file

However, I don't know how where generated the other binary files in this directory

**Comment by CarlosAlbertoEnciso** [comment]
@aurelien35 I can check my notes about the steps used to generate the other binary files.

**Comment by CarlosAlbertoEnciso** [comment]
@aurelien35 This is the information extracted from my notes.

Source file: `test.cpp`
```
using INTPTR = const int *;
int foo(INTPTR ParamPtr, unsigned ParamUnsigned, bool ParamBool) {
  if (ParamBool) {
    typedef int INTEGER;
    const INTEGER CONSTANT = 7;
    return CONSTANT;
  }
  return ParamUnsigned;
}
```
Command line used to generate the binary files:

`-- Linux`
```
clang -c -w -g -gdwarf-4 -O0 test.cpp -o test-dwarf-clang.o

g++ -c -w -g -O0 test.cpp -o test-dwarf-gcc.o
```

`-- Windows`
```
clang++.exe --target=x86_64-windows -c -w -g -gcodeview test.cpp -o test-codeview-clang.o

cl.exe /nologo /EHsc /Od /Z7 /Fotest-codeview-msvc.o /c test.cpp
cl.exe /nologo /EHsc /Od /Zi /Fotest-codeview-pdb-msvc.o /Fdtest-codeview-pdb-msvc-.pdb /c test.cpp
```

**Comment by aurelien35** [comment]
@CarlosAlbertoEnciso I've updated the README.md file with your notes

**Comment by CarlosAlbertoEnciso** [comment]
@pogo59 Do you have any comments on this patch? Thanks

File: llvm/unittests/DebugInfo/LogicalView/CodeViewReaderTest.cpp [hunk 1]
```diff
@@ -193,6 +196,72 @@ void checkElementPropertiesMsvcCodeview(LVReader *Reader) {
   EXPECT_EQ(Lines->size(), 0x0eu);
 }
 
+// Check the logical elements basic properties (MSVC library - Codeview).
+void checkElementPropertiesMsvcLibraryCodeview(LVReader *Reader) {
+  LVScopeRoot *Root = Reader->getScopesRoot();
+  LVScopeCompileUnit *CompileUnit =
+      static_cast<LVScopeCompileUnit *>(getFirstScopeChild(Root));
+  LVScopeFunction *Function =
```
Comment (CarlosAlbertoEnciso) [inline llvm/unittests/DebugInfo/LogicalView/CodeViewReaderTest.cpp:204]: Same check here for `CompileUnit` and `Function`.

**Comment by pogo59** [comment]
> @pogo59 Do you have any comments on this patch? Thanks

No, it's fine.

**Comment by aurelien35** [comment]
@CarlosAlbertoEnciso Thanks.

What's next now? Should I do something or just wait for an admin to merge this patch?

**Comment by CarlosAlbertoEnciso** [comment]
@aurelien35 Do you commit access?

**Comment by aurelien35** [comment]
@CarlosAlbertoEnciso no, I don't have commit access

**Comment by CarlosAlbertoEnciso** [comment]
@aurelien35 Did you added the suggested checks? I can commit for you.

File: llvm/unittests/DebugInfo/LogicalView/CodeViewReaderTest.cpp [hunk 2]
```diff
@@ -193,6 +196,72 @@ void checkElementPropertiesMsvcCodeview(LVReader *Reader) {
   EXPECT_EQ(Lines->size(), 0x0eu);
 }
 
+// Check the logical elements basic properties (MSVC library - Codeview).
+void checkElementPropertiesMsvcLibraryCodeview(LVReader *Reader) {
+  LVScopeRoot *Root = Reader->getScopesRoot();
```
[the remark this answers is not in the document]
Comment (CarlosAlbertoEnciso) [reply llvm/unittests/DebugInfo/LogicalView/CodeViewReaderTest.cpp:201]: I will add this check in a separated patch.

File: llvm/unittests/DebugInfo/LogicalView/CodeViewReaderTest.cpp (hunk 1 above)
Comment (CarlosAlbertoEnciso) [reply llvm/unittests/DebugInfo/LogicalView/CodeViewReaderTest.cpp:204]: I will add this check in a separated patch.

File: llvm/unittests/DebugInfo/LogicalView/CodeViewReaderTest.cpp [hunk 3]
```diff
@@ -193,6 +196,72 @@ void checkElementPropertiesMsvcCodeview(LVReader *Reader) {
...
+  EXPECT_EQ(Function->lineCount(), 14u);
+  EXPECT_EQ(Function->scopeCount(), 1u);
+  EXPECT_EQ(Function->symbolCount(), 3u);
+  EXPECT_EQ(Function->typeCount(), 0u);
+  EXPECT_EQ(Function->rangeCount(), 1u);
+
+  const LVLocations *Ranges = Function->getRanges();
+  ASSERT_NE(Ranges, nullptr);
+  ASSERT_EQ(Ranges->size(), 1u);
+  LVLocations::const_iterator IterLocation = Ranges->begin();
```
[the remark this answers is not in the document]
Comment (CarlosAlbertoEnciso) [reply llvm/unittests/DebugInfo/LogicalView/CodeViewReaderTest.cpp:223]: I will add this check in a separated patch.

**Pull request merged**